packages/trpc/server/routers/viewer/teams/lazyLoadMembers.handler.ts

```diff
@@ -0,0 +1,194 @@
+        select: {
+          externalId: true,
+        },
+      },
+    },
+  },
+});
+
+export const lazyLoadMembersHandler = async ({ ctx, input }: LazyLoadMembersHandlerOptions) => {
+  const { prisma } = ctx;
+  const { cursor, limit, teamId, searchTerm } = input;
+
+  const canAccessMembers = await checkCanAccessMembers(ctx, teamId);
+
+  if (!canAccessMembers) {
+    throw new TRPCError({
+      code: "UNAUTHORIZED",
+      message: "You are not authorized to access this team",
+    });
+  }
+
+  const teamMembers = await prisma.membership.findMany({
+    where: {
+      teamId,
+      ...(searchTerm && {
+        user: {
+          OR: [
+            {
+              email: {
+                contains: searchTerm,
+              },
+            },
+            {
+              username: {
+                contains: searchTerm,
+              },
+            },
+          ],
+        },
+      }),
+    },
+    select: {
+      id: true,
+      role: true,
+      accepted: true,
+      user: {
+        select: userSelect,
+      },
+    },
+    cursor: cursor ? { id: cursor } : undefined,
+    take: limit + 1, // We take +1 as itll be used for the next cursor
+    orderBy: {
+      id: "asc",
+    },
+  });
+
+  let nextCursor: typeof cursor | undefined = undefined;
+  if (teamMembers && teamMembers.length > limit) {
+    const nextItem = teamMembers.pop();
+    nextCursor = nextItem?.id;
+  }
+
+  const members = await Promise.all(
+    teamMembers.map(async (member) => ({
+      ...member,
+      user: await UserRepository.enrichUserWithItsProfile({
+        user: member.user,
+      }),
+    }))
+  );
+
+  const membersWithApps = members.map((member) => {
+    const { credentials, profile, ...restUser } = member.user;
+    return {
+      ...restUser,
+      username: profile?.username ?? restUser.username,
+      role: member.role,
+      profile: profile,
+      organizationId: profile?.organizationId ?? null,
+      organization: profile?.organization,
+      accepted: member.accepted,
+      disableImpersonation: member.user.disableImpersonation,
+      bookerUrl: getBookerBaseUrlSync(profile?.organization?.slug || ""),
+      connectedApps: credentials?.map((cred) => {
+        const appSlug = cred.app?.slug;
+        let appData = appDataMap.get(appSlug);
+
+        if (!appData) {
+          appData = getAppFromSlug(appSlug);
+          appDataMap.set(appSlug, appData);
+        }
+
+        const isCalendar = cred?.app?.categories?.includes("calendar") ?? false;
+        const externalId = isCalendar ? cred.destinationCalendars?.[0]?.externalId : null;
+        return {
+          name: appData?.name ?? null,
+          logo: appData?.logo ?? null,
+          app: cred.app,
+          externalId: externalId ?? null,
+        };
+      }),
+    };
+  });
+
+  return { members: membersWithApps, nextCursor };
+};
+
+const checkCanAccessMembers = async (ctx: LazyLoadMembersHandlerOptions["ctx"], teamId: number) => {
```

Comment:
Authorization check

Reply:
This is almost similar to shouldHideMembers of teams/get.handler.ts.

Can we reuse that please. This is critical logic and we should followup with unit tests later.